### ISSUE TITLE
question about dataset

### ISSUE THREAD

**Issue by axeasy** [snoop2head/DotNeuralNet#1]
Hello, I downloaded the corresponding dataset according to the readme, but encountered an issue while running the train.exe file. Could you please provide specific files that need to be stored in the dataset

**Comment by snoop2head** [comment]
@axeasy 
Can you please specify the error? Let's begin with which file you ran in the beginning. train.exe does not exist in this repo btw

**Comment by axeasy** [comment]
Sorry, I typed the wrong word. It's a train.py file. The first issue I encountered was that I saw this file pointing to dataset.py, so I downloaded four datasets from the dataset folder and put them in the corresponding folder. However, an error occurred during training because the dataset.py file contained :self. kaggle. path, self. angelina_path, and self. dsbi-path, each pointing to a folder called "cropped image", but none of these datasets were included

**Comment by snoop2head** [comment]
Oh, I see.
- Make a folder with bash commands `mkdir cropped_image` or with graphic user interface.
- Crop region of interest (=square / rectangle area of the braille dots) from each of the dataset and place in the corresponding `cropped_image` folder.
- When you are cropping and saving in the `cropped_image` folder, make sure that the file basename formats be like "010000.jpg" or "100100.jpg", so that the labels are readable with functions at https://github.com/snoop2head/DotNeuralNet/blob/f4880ebba37decd4e23eeec08d550c96dc662fcc/src/dataset.py#L83-L103

It's a bit of nuisance but necessity for making braille classifier. **But if you want to skip the training process, just use yolo models at [weights directory](https://github.com/snoop2head/DotNeuralNet/tree/main/weights) which I made detection/classification models public.**

What is your purpose of using the project? Training from scratch or building an application based on the available model?

**Comment by axeasy** [comment]
Thank you for your help.First of all, I have a question. Based on the conversation just now, I want to know whether the images stored in cropped images are cropped from their parent dataset. For example, should 'dataset/dsbi/cropped images' contain braille squares cropped from any image in the DSBI dataset? Secondly, I want to learn and explore the process of Braille image recognition through this project and use it as a basis to complete my graduation thesis.So I want to start from the training process until the recognition is completed

**Comment by axeasy** [comment]
I put the cropped images from ‘a’ to ‘z’ under the ‘cropped images’ folder in the angelinadataset folder and named them in the format of ‘010000’, and put the cropped images from ‘a’ to ‘z’ under the ‘cropped images’ folder in the ‘DSBI’ and ‘kaggle’ folders, and named them in the format of ‘a,b,c,…,z’. Now I can start training, but I want to know if I train in this way, can a dataset only have 64 images at most, because there are only 64 types of Braille.

**Comment by snoop2head** [comment]
@axeasy Sorry, I've been away because of rebuttal for other conferences. 

The project's purpose was to create a model robust to both book background and natural background which coerced me to create multi-label classification model. However, to the best of my knowledge, `braille_natural` dataset only has region of interest (RoI) positions but does not provide the specific braille label. In contrast, `DSBI` or `AngelinaDataset` contain both RoI bounding box and human labeled classes. So the goal of creating multi-label classification model is to recognize 2^6 types of braille and make the model to function as a pseudo-labeler in natural scene background `braille_natural` dataset.

After pseudo-labeling with the created model, you can move on to creating a detection model which is documented at [yolov8 documentation](https://github.com/ultralytics/ultralytics?tab=readme-ov-file#documentation). But you may skip the pseudo-labeling part if your model is going to receive braille inputs from books only.

**Comment by axeasy** [comment]
I'm sorry, I haven't been working recently because of the holiday. Can we add a contact method on other platforms to communicate?

**Comment by axeasy** [comment]
I would like to ask if the train.py file in your project is used for training target detection.

**Comment by axeasy** [comment]
I converted the dsbi dataset into yolo format for training, but the accuracy after training is only between 0.002 and 0.02. According to my observation, the obj_loss of training is particularly large, but I did not get any improvement after modifying the anchorbox. How did you train the yolo weights?

**Comment by HuangruiChu** [comment]
I also wonder this part. The dataset in this project is just for the purpose of "classification". The model and the dataset in the project do not have any info about the "detection" part.

**Comment by HuangruiChu** [comment]
https://github.com/Mcblaze95/braille_recognition/commits/main/yolov8_braille.pt    commited on May 18,2023
https://github.com/nekohamja/braillify/commits/main/yolov8_braille.pt commited on May 16, 2023

It seems that these could be copied from you. So you should have the record of training for these checkpoints.

**Comment by snoop2head** [comment]
@HuangruiChu 

### Classification
You are right about dataset is for the purpose of classification, but to clarify, it is for pseudo-labeling purpose on ["Natural Scene Braille" dataset](https://onlinelibrary.wiley.com/doi/10.1155/2022/7201775). The dataset does not contain labeled symbols, but only bounding boxes which led me to label it separately. After training on disjoint training data, it inferences cropped bounding boxes of "Natural Scene Braille" dataset so that it will finally have its labels. That is the intent of the classification.

### Training Object Detection
The training log of object detection is attached on the [project's README](https://github.com/snoop2head/DotNeuralNet/tree/main?tab=readme-ov-file#logs). Training code for object detection relies on [yolov5](https://github.com/ultralytics/yolov5) and [yolov8](https://github.com/ultralytics/ultralytics) which means that the repository won't contain a related code. 

### Authorship
I think you may be pointing out authorship, and thanks for pointing out. I am not sure why they haven't cited my repository though.

**Comment by axeasy** [comment]
Appreciated a lot about your reply and work, I've cited your repository in my thesis

**Comment by snoop2head** [comment]
@axeasy I am glad it helped 😅 I think I was quite busy to reply to you promptly but thank you for your understanding!

**Comment by luke2023** [comment]
@snoop2head 
Thank you for your work.
 I am now in the step of pseudo labeling but In the pseudo_labeling.py, I could not find the function of 'crop_image' in the utils

 the crop function seems to be missing 
from utils import get_label, get_image, crop_image

Could you please clarify that?
Thanks in advance
